sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OpenTelemetrySdkAutoConfiguration.java

```diff
@@ -25,25 +25,30 @@
  */
 public final class OpenTelemetrySdkAutoConfiguration {
 
-  private static final Resource RESOURCE = buildResource();
+  private static volatile Resource resource = Resource.empty();
 
-  /** Returns the automatically configured {@link Resource}. */
+  /**
+   * Returns the automatically configured {@link Resource} after one of {@code initialize()} methods
+   * has been called. Prior to that, this method will always return an {@linkplain Resource#empty()
+   * empty resource}.
+   */
   public static Resource getResource() {
-    return RESOURCE;
+    return resource;
   }
```

Comment:
I really dislike this part of this PR: now that we allow using custom `ConfigProperties` impl there's a clear temporal dependency between `getResource()` and `initialize()` - which I guess sort of makes sense but I still don't like it.

Reply:
This is only the case because of the `getResource()` method. Get rid of this method and the `initialize` methods can operate without any side affects in `OpenTelemetrySdkAutoconfiguration`.

To me, `getResource` feels awkward here. Feels like it could live on `OpenTelemetrySdk` or `OpenTelemetry` instead.

Reply:
We can't put it into `OpenTelemetry` because the resource is a part of SDK and not API, but `OpenTelemetrySdk` sounds good.
@anuraaga @jkwatson was there any reason for not adding it to the `OpenTelemetrySdk`?

Reply:
There is a need to be able to get a handle to the Resource that would be autoconfigured, then modify that and use it to build an SDK. So, it would be very cumbersome to have to build an SDK, then get the resource, modify it and rebuild a new SDK.

Reply:
But, I agree that this API is not good, as-is. Perhaps we should split out the Resource autoconfiguration into it's own class. Something like `OpenTelemetryResource.initialize()`

Reply:
Done, sort of - I've added an `OpenTelemetryResourceAutoConfiguration` class that borrows a few patterns from `GlobalOpenTelemetry` and holds the global auto-configured resource.